File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IAudiosLocalView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.Audio;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;
import dev.ragnarok.fenrir.upload.Upload;

public interface IAudiosLocalView extends IMvpView, IErrorView, IAccountDependencyView {
    void displayList(List<Audio> audios);

    void notifyItemChanged(int index);

    void notifyItemRemoved(int index);

    void notifyListChanged();

    void notifyUploadItemsAdded(int position, int count);

    void notifyUploadItemRemoved(int position);

    void notifyUploadItemChanged(int position);

    void notifyUploadProgressChanged(int position, int progress, boolean smoothly);

    void setUploadDataVisible(boolean visible);

    void displayUploads(List<Upload> data);

    void notifyUploadDataChanged();

    void displayRefreshing(boolean refreshing);

    void checkPermission();
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/IAudiosLocalView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.Audio
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView
import dev.ragnarok.fenrir.upload.Upload

interface IAudiosLocalView : IMvpView, IErrorView, IToastView, IAccountDependencyView {
    fun displayList(audios: List<Audio>)
    fun notifyItemChanged(index: Int)
    fun notifyItemRemoved(index: Int)
    fun notifyListChanged()
    fun notifyUploadItemsAdded(position: Int, count: Int)
    fun notifyUploadItemRemoved(position: Int)
    fun notifyUploadItemChanged(position: Int)
    fun notifyUploadProgressChanged(position: Int, progress: Int, smoothly: Boolean)
    fun setUploadDataVisible(visible: Boolean)
    fun displayUploads(data: List<Upload>)
    fun displayRefreshing(refreshing: Boolean)
    fun checkPermission()
}